modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -370,8 +370,21 @@ public void processOpts() {
             documentationProvider = DocumentationProvider.NONE;
             annotationLibrary = AnnotationLibrary.NONE;
             useJakartaEe=true;
+            useBeanValidation = false;
+            performBeanValidation = false;
+
             additionalProperties.put(USE_JAKARTA_EE, useJakartaEe);
+            additionalProperties.put(USE_BEANVALIDATION, useBeanValidation);
+            additionalProperties.put(PERFORM_BEANVALIDATION, performBeanValidation);
+            additionalProperties.put(DOCUMENTATION_PROVIDER, documentationProvider.toCliOptValue());
+            additionalProperties.put(documentationProvider.getPropertyName(), true);
+            additionalProperties.put(ANNOTATION_LIBRARY, annotationLibrary.toCliOptValue());
+            additionalProperties.put(annotationLibrary.getPropertyName(), true);
+
             applyJakartaPackage();
+
+            LOGGER.warn("For Spring HTTP Interface following options are disabled: documentProvider, annotationLibrary, useBeanValidation, performBeanValidation. "
```

Comment:
I think the annotation library should still be used...

The project I work on, we had the interface option but rely on the annotations on the interfaces to provide the Springdoc content for the document page that gets generated.

I suspect that disabling that option might use the incorrect annotations?

Did no samples update from this change?

Reply:
> The project I work on, we had the interface option but rely on the annotations on the interfaces to provide the Springdoc content for the document page that gets generated.


hmmm. My point was that since this is a client library then it doesn't require any documentation meta data. If you think that here we need support for springdoc/swagger2 then I can add it

> Did no samples update from this change?

nope it was disabled by default but with ability to override it via config - this is what happened for user. Now there is no ability to override those configurations

Reply:
> > The project I work on, we had the interface option but rely on the annotations on the interfaces to provide the Springdoc content for the document page that gets generated.
> 
> hmmm. My point was that since this is a client library then it doesn't require any documentation meta data. If you think that here we need support for springdoc/swagger2 then I can add it
> 
> > Did no samples update from this change?
> 
> nope it was disabled by default but with ability to override it via config - this is what happened for user. Now there is no ability to override those configurations

Sorry I think I misunderstood - this is for request clients as opposed to servers? Using Spring HTTP interface? 

If so, then makes sense to me. I guess the logger comment `Spring HTTP Interface` I assumed it was interface only server stubs

Reply:
> Sorry I think I misunderstood - this is for request clients as opposed to servers? Using Spring HTTP interface?

This is a client library that works based on Spring's WebClient. Since you already have source with have respective annotations (`@RequestBody`, `@RequstParam`, `@PathParam`) I thought that there is no need to generate swagger meta annotations since they don't bring any value